samples/readme.md

```diff
@@ -0,0 +1,18 @@
+# Plug-in Creation for Performance SDK
+
+## Implement a Simple Data Source
+
+Sample source code: ./SimpleDataSource
+
+To create a simple data source, perform the following:
+
+1) Create a public class that implements the abstract class CustomDataSourceBase.
+2) Create a public class that implements the abstract class CustomDataProcessorBase.
+3) Create one or more data tables classes. These classes must:
+   - Be public and static.
+   - Be decorated with TableAttribute.
+   - Expose a static public field or property named TableDescriptor of type TableDescriptor which provides information about the table.
+
+The custom data source class will create a new instance of the custom data processor class when ProcessAsyncCore is called.
```

Comment:
This sentence is confusing for multiple reasons:

1) "custom data source" is overloaded, and can refer to either the deriving class mentioned in step 1 above, or the abstract CustomDatasource. To fix this, maybe say "Your custom data source class" instead of "the."

2) Saying "will create" implies it happens auto-magically (i.e. the SDK is taking care of it). Since the processor is being manually instantiated, perhaps clarify this with "must create a new instance of your custom data processor in its constructor"

3) "when ProcessAsyncCore is called" implies the following logic:
    SDK -> Processor.ProcessAsyncCore -> Source() -> Processor()
i.e. it sounds like the processor gets called before it's instantiated. To fix this, we should re-organize this sentence. My suggestion (which may be incorrect because I'm not too familiar with the execution order) is:
"When the SDK needs to process a file your custom data source supports, it will obtain an instance of your custom data processor through the custom data source's constructor and pass the file to its `ProcessAsyncCore` method. Thus, your custom data source must create and return a new instance of your custom data processor in its constructor."